File: 2870-minimum-number-of-operations-to-make-array-empty/2870-minimum-number-of-operations-to-make-array-empty.py
class Solution(object):
    def minOperations(self, nums):
        mp = {}
        for n in nums:
            mp[n] = mp.get(n, 0) + 1
        count = 0

        for key, value in mp.items():
            if value == 1:
                return -1
            count += value//3
            if value % 3 != 0:
                count += 1
            
        return count
        

File: 2870-minimum-number-of-operations-to-make-array-empty/2870-minimum-number-of-operations-to-make-array-empty.cpp
class Solution {
public:
    int minOperations(vector<int>& nums) {
        unordered_map<int,int> mp;
        for(auto it : nums) mp[it]++;
        int count = 0;
        
        for(auto it : mp) {
            int x = it.second;
            if(x == 1) return -1;
            count += x / 3 + (x % 3 != 0); 
        }

        return count;
    }
};